uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Listener/AbstractListener.php

```diff
@@ -0,0 +1,25 @@
+<?php
+  abstract class CRM_HRCore_Listener_AbstractListener {
+
+    protected $objectClass;
+    protected $object;
+
+    public function __construct(&$object) {
+      $this->object = &$object;
+    }
+
+    protected function canHandle() {
+      return $this->object instanceof $this->objectClass;
+    }
+
+    protected function isExtensionEnabled($key) {
```

Comment:
isExtensionEnabled does not fit into this class, it has nothing to do with it,  I understand that you added it to avoid code duplication  but I think the effect of adding it to this class is worse that the effect of duplicating the code. 

I would say let's duplicate it for now until we find a more appropriate place to put such methods.